### PR TITLE
Update UG and DG with Delete and Stats features, Fix some PED Documentation Bugs

### PR DESCRIPTION
**Pull request by butteredyakiimo**
- Make some PED bug fixes (see pic in tele)
- Standardise notations used in UG
- Add some planned enhancements
- Edit use cases and user stories
- Modify manual testing in DG
- Add in implementation details for Delete and Stats

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -342,11 +405,55 @@ We are also planning to shorten some input parameters when adding or editing a f
 | availability/nil          | avail/nil    |
 | animalType/               | type/        |
 
+With this, the command `add n/Jerry Tan p/98765412 e/jerry123@example.com housing/HDB avail/true` will be a valid add
+command. 
+
+### Reduce Coupling Between Availability and Animal Type
+Currently, the `animalType` field also contains information about the availability of a fosterer.<br>
+e.g. if the `animalType` field of a fosterer is `current.Dog`, it suggests that the fosterer is available, and is 
+fostering a dog. However, the user will still need to enter the `availability` field as `available`. <br>
```
Comment (nananakx-x) [inline docs/DeveloperGuide.md]: you mean "NotAvailable" here? since it is current.Dog

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -342,11 +405,55 @@ We are also planning to shorten some input parameters when adding or editing a f
...
+
+Currently, we only allow the assignment of one fosterer to one animal. To improve this, we can make the `ANIMAL_NAME`,
+`ABLE_ANIMAL_TYPE` and `CURRENT_ANIMAL_TYPE` fields to be maintained as lists instead. This will allow one fosterer to be associated with more than one animal.
+
 ### Case-sensitivity of Inputs
-Currently, the fields and parameters for housing, availability and animal type are case-sensitive. An enhancement to this would be to make these fields and their parameters case-insensitive to improve the user experience. Moreover, the aforementioned enhancements to shorten the command format will also be case-insensitive.
+Currently, the fields and parameters for housing, availability and animal type are case-sensitive. An enhancement to this would be to make these fields and their parameters case-insensitive to improve the user experience. Moreover, the aforementioned enhancements will also be case-insensitive.
+
+### Allow Symbols in Name
+Currently, names in Foster Family must be alphanumeric. However, this excludes certain legal names that have other
```
Comment (nananakx-x) [inline docs/DeveloperGuide.md:440]: wait i thought we are also adding the point about how name should not accept numeric string

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -656,29 +758,21 @@ testers are expected to do more *exploratory* testing.
...
 
-1. _{ more test cases …​ }_
 
 ### Deleting a person
 
-1. Deleting a person while all persons are being shown
+1. Deleting a person while all fosterers are being shown
 
-   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+   1. Prerequisites: List all persons using the `list` or `find` command. At least 6 fosterers in the list.
```
Comment (nananakx-x) [inline docs/DeveloperGuide.md]: why need at least 6 fosterers in the list to delete multiple...?

**Review by nananakx-x — COMMENTED** [review]
Did not really read the UG changes in detail, but will definitely do so when I am adding on later! We should merge this and later check if we still have formatting errors and I can change them later :)

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (butteredyakiimo) [reply docs/DeveloperGuide.md]: sorry i meant able.Dog HAHAH so the fosterer is available! good catch

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (butteredyakiimo) [reply docs/DeveloperGuide.md:440]: done!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (butteredyakiimo) [reply docs/DeveloperGuide.md]: sorry it was because of my test case, ill change it to 3 instead

**Review by nananakx-x — APPROVED** [review]
lgtm

**Pull request merged**